src/main/kotlin/org/wycliffeassociates/otter/jvm/recorder/device/AudioRecorder.kt

```diff
@@ -0,0 +1,81 @@
+package org.wycliffeassociates.otter.jvm.recorder.device
+
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import org.wycliffeassociates.otter.common.device.IAudioRecorder
+import javax.sound.sampled.AudioFormat
+import javax.sound.sampled.AudioSystem
+
+class AudioRecorder : IAudioRecorder{
+
+    private val monitor = Object()
+
+    @Volatile
+    private var stop = false
+
+    companion object {
+        val SAMPLE_RATE = 44100F // Hz
+        val SAMPLE_SIZE = 16 // bits
+        val CHANNELS = 1
+        val SIGNED = true
+        val BIG_ENDIAN = false
+        val FORMAT = AudioFormat(
+            SAMPLE_RATE,
+            SAMPLE_SIZE,
+            CHANNELS,
+            SIGNED,
+            BIG_ENDIAN
+        )
+        val BUFFER_SIZE = 1024
+    }
+
+    private var line = AudioSystem.getTargetDataLine(FORMAT)
+    private val audioByteObservable = PublishSubject.create<ByteArray>()
+    private val recordingStream = Observable.fromCallable {
+        val byteArray = ByteArray(BUFFER_SIZE)
+        var totalRead = 0
+            while (true) {
+                if(line.isOpen || line.available() > 0) {
+                    totalRead += line.read(byteArray, 0, byteArray.size)
+                    audioByteObservable.onNext(byteArray)
+                } else {
+                    synchronized(monitor) {
+                        monitor.wait()
+                    }
+                }
+                if (stop) {
+                    line.close()
+                    break
+                }
+            }
+    }.subscribeOn(Schedulers.io())
+    .subscribe()
+
+    @Synchronized //Synchronized so as to not subscribe to multiple streams on quick multipress
+    override fun start() {
+        line.open(FORMAT)
+        line.start()
```

Comment:
I've been looking through the java code, and I can't find a function body for start() in either DataLine or TargetDataLine???

Reply:
They're both interfaces implemented by the line provided by the AudioSystem. If you search back far enough it'll likely be C code that's platform dependent.